frontend/pkg/frontend/context.go

```diff
@@ -137,3 +138,17 @@ func SubscriptionStateFromContext(ctx context.Context) (arm.RegistrationState, e
 	}
 	return subscriptionState, nil
 }
+
+func ContextWithTenantID(ctx context.Context, sub arm.Subscription) context.Context {
```

Comment:
Why are we passing the whole `arm.Subscription` struct here just to pluck out the `TenantId` string?

Alternatively, would it make more sense to stash the entire `arm.Subscription` struct in the context?  That could replace both `ContextWithTenantID` and `ContextWithSubscriptionState`.

Reply:
Just for typing/prevent user error sending the wrong string. `SubscriptionState` has its own unique type already to mitigate the issue, but `TenantID` is just a vanilla string. Definitely no harm just folding the two together though, I'll do that.

Reply:
To your point about typing, I wonder if a type alias in the `api/arm` package would be useful.
```
type TenantID string
```